connexion/decorators/response.py

```diff
@@ -56,6 +56,14 @@ def validate_response(self, data, status_code, headers, url):
                 msg = ("Keys in header don't match response specification. "
                        "Difference: {0}").format(pretty_list)
                 raise NonConformingResponseHeaders(message=msg)
+            # validate each of the existing keys.
+            for key, schema in response_definition.get("headers").items():
+                header_validator = ResponseHeaderValidator(schema)
+                for data in headers.getlist(key):
```

Comment:
The response object here isn't even of type `ConnexionResponse`.  I think this is somewhat premature.

Nevertheless, https://github.com/zalando/connexion/pull/464 is up for your review.